packages/did-core-test-server/suites/did-spec/did-producer.js

```diff
@@ -0,0 +1,70 @@
+const utils = require('./utils');
+const jsonMediaTypes = ['application/did+ld+json', 'application/did+json'];
+const deepEqual = require('deep-equal')
+
+const generateDidProducerTests = ({did, resolutionResult}) => {
+  const {didDocument} = resolutionResult;
+  const contentType = resolutionResult.didDocumentMetaData['content-type'];
+
+  it('6.1 Production and Consumption - A conforming producer MUST take a ' +
+    'DID document data model and a representation-specific entries map as ' +
+    'input into the production process. The conforming producer MAY accept ' +
+    'additional options as input into the production process.', async () => {
+      if(jsonMediaTypes.includes('application/did+ld+json')) {
+        const context = didDocument['@context'];
+        if(typeof context === 'string') {
+          expect(context).toBe('https://www.w3.org/ns/did/v1');
+        } else if(Array.isArray(context)) {
+          expect(context[0]).toBe('https://www.w3.org/ns/did/v1');
+          reserializedContext = JSON.parse(JSON.stringify(context));
+          expect(deepEqual(context, reserializedContext)).toBe(true);
+        } else {
+          throw new Error('Invalid @context value '+ context);
+        }
+      }
+  });
+
+  it('6.1 Production and Consumption - A conforming producer MUST ' +
+    'serialize all entries in the DID document data model, and the ' +
+    'representation-specific entries map, that do not have explicit ' +
+    'processing rules for the representation being produced using only the ' +
+    'representation\'s data type processing rules and return the ' +
+    'serialization after the production process completes.', async () => {
+      if(jsonMediaTypes.includes(contentType)) {
+        reserializedDidDocument = JSON.parse(JSON.stringify(didDocument));
+        expect(deepEqual(didDocument, reserializedDidDocument)).toBe(true);
+      } else {
+        throw new Error('Unknown producer for content-type: '+ contentType);
+      }
+  });
+
+  it('6.1 Production and Consumption - A conforming producer MUST return ' +
+    'the Media Type string associated with the representation after the ' +
+    'production process completes.', async () => {
+      expect(typeof contentType === 'string').toBe(true);
```

Comment:
This feels insufficient.. How does this test the statement above?

Reply:
should be `contentType === application/did+json`, or similar.

Reply:
The normative statement is just saying that a contentType has to exist and the spec defines contentType as being a string. That's what the test is testing.

> should be contentType === application/did+json, or similar.

Remember that this test runs /for all producers/. So, if the test was the above... any non-JSON representation would fail the test, which is not what we want. We just want to make sure producers are returning a media type string for a representation.

Reply:
The normative statement says much more here than just "the content type MUST be a string", which is what the code is testing....

Reply:
> The normative statement says much more here than just "the content type MUST be a string", which is what the code is testing....

What test would you write for this normative statement?